config/config.go

```diff
@@ -826,8 +828,12 @@ func (c *RemoteWriteConfig) UnmarshalYAML(unmarshal func(interface{}) error) err
 	httpClientConfigAuthEnabled := c.HTTPClientConfig.BasicAuth != nil ||
 		c.HTTPClientConfig.Authorization != nil || c.HTTPClientConfig.OAuth2 != nil
 
-	if httpClientConfigAuthEnabled && c.SigV4Config != nil {
-		return fmt.Errorf("at most one of basic_auth, authorization, oauth2, & sigv4 must be configured")
+	if httpClientConfigAuthEnabled && (c.SigV4Config != nil || c.AzureADConfig != nil) {
```

Comment:
Just an idea: I wonder if it is worth counting number of auth options enabled rather than multiple if blocks? That might be easier to understand but I am not sure.

Reply:
I thought about this, however, it didn't seem pretty straightforward and felt might get more confusing going forward if some new auth was added.